src/status_im2/contexts/onboarding/create_profile/view.cljs

```diff
@@ -37,40 +38,54 @@
   [s]
   (cond
     (or (= s nil) (= s ""))      nil
-    (has-emojis s)               (i18n/label :t/are-not-allowed {:check (i18n/label :t/emojis)})
     (has-special-characters s)   (i18n/label :t/are-not-allowed
                                              {:check (i18n/label :t/special-characters)})
     (string/ends-with? s "-eth") (i18n/label :t/ending-not-allowed {:ending "-eth"})
     (has-common-names s)         (i18n/label :t/are-not-allowed {:check (i18n/label :t/common-names)})
+    (has-emojis s)               (i18n/label :t/are-not-allowed {:check (i18n/label :t/emojis)})
     :else                        nil))
 
+(defn button-container
+  [keyboard-shown? children]
+  (if keyboard-shown?
+    [blur/view
+     {:blur-type :dark
+      :style     style/blur-button-container}
+     children]
+    [rn/view {:style style/view-button-container}
+     children]))
+
 (defn page
   [{:keys [image-path display-name color]}]
-  (let [full-name             (reagent/atom display-name)
-        validation-msg        (reagent/atom (validation-message @full-name))
-        on-change-text        (fn [s]
-                                (reset! validation-msg (validation-message s))
-
-                                (reset! full-name s))
-        custom-color          (reagent/atom (or color c/profile-default-color))
-        profile-pic           (reagent/atom image-path)
-        on-change-profile-pic #(reset! profile-pic %)
-        on-change             #(reset! custom-color %)]
+  (reagent/with-let [full-name             (reagent/atom display-name)
+                     keyboard-shown?       (reagent/atom false)
```

Comment:
I experimented with hooks for a while but I found with-let less brittle. Hooks kept breaking saying it had rendered with a different number of hooks error.  Perhaps, I had not configured everything correctly though.
Do we have a preference for when we should use hooks vs with-let? 🤔

Reply:
`with-let` something new, we've never used it before, we have a few places where we use a keyboard hook, it seems like less code and simpler to use, but i don't know :f> usage is also not great and better to do not use it if possible, so here seems like we don't need it, ok for now lets just move listeners code outside , so make code little bit cleaner , thanks

Reply:
> Hooks kept breaking saying it had rendered with a different number of hooks error.

It had happened to me.
There are some rules you need to follow in order to correctly use hooks. Sometimes, I think they look like a "patch" on React due to its implementation.

- They need to be called in the same order across re-renders. If not followed, the Hook state could break.

- If you are using a hook in the render function, make sure you're using it across all re-renders, this error happened to me when I tried to conditionally use a hook, for example, calling it  inside an `if` or `when`, since the body is evaluated according to the test parameter:

```clojure
(when test hook-call)
```

If test changes, your component will render a different number of hooks than the last render. Solution:

```clojure
(let [hook-val hook-call] ;; <- this is always callled, no matter what test is
  (when test hook-val)) 
```

if you'd like to take a look at the docs: https://legacy.reactjs.org/docs/hooks-rules.html

Reply:
For this component, I think both hook and with-let approaches work. 👍

This component looks good to me, I'm not against functional components, but I prefer this one since there is no need to wrap it in a new component only to add `[f:>]`.

Reply:
Thanks Ulises, will address those points.
I'm familiar with how hooks work in react, the issue here is that hooks have to be handled quite particularly where as with-let is more robust particularly for my simple use case. So I prefer the use of with-let here